tests/pfcwd/files/pfcwd_helper.py

```diff
@@ -183,19 +188,87 @@ def parse_vlan_list(self):
             temp_ports (dict): port info constructed from the vlan interfaces
         """
         temp_ports = dict()
-        vlan_members = self.vlan_info[self.vlan_info.keys()[0]]['members']
+        vlan_details = self.vlan_info.values()[0]
+        vlan_members = vlan_details['members']
+        vlan_type = vlan_details.get('type')
+        vlan_id = vlan_details['vlanid']
+        rx_port = self.pfc_wd_rx_port if isinstance(self.pfc_wd_rx_port, list) else [self.pfc_wd_rx_port]
+        rx_port_id = self.pfc_wd_rx_port_id if isinstance(self.pfc_wd_rx_port_id, list) else [self.pfc_wd_rx_port_id]
```

Comment:
Is this conversion done in the testcase as well in send traffic method? How was this working before?

Reply:
This is because the traffic verification methods all assume the rx port and rx port id are all lists:
https://github.com/Azure/sonic-mgmt/blob/4ab629f2bab4eb5585eb15d44cd2866346f25d58/tests/pfcwd/test_pfcwd_function.py#L429

Reply:
I see it now. It worked all the while because we call parse_pc_list() and parse_intf_list() before parsing vlans. so pfc_wd_rx_port is already populated by them and they always populate a list. You have the same logic in parse_vlan_sub_intf_list where you are populating a list for the pfc_wd_rx_port, so this check isn't needed but no harm in having it as well